ql/src/java/org/apache/hadoop/hive/ql/io/parquet/serde/ArrayWritableObjectInspector.java

```diff
@@ -222,29 +222,47 @@ public Object setStructFieldData(Object struct, StructField field, Object fieldV
   }
 
   @Override
-  public boolean equals(Object obj) {
-    if (obj == null) {
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    ArrayWritableObjectInspector that = (ArrayWritableObjectInspector) o;
+
+    if (isRoot != that.isRoot) {
```

Comment:
hint: can we merge all these three "if" clauses?  The same goes to ```equals``` method. You can break lines before "||" for better view.